src/plugins/gui/providers/ioniaProvider.ts

```diff
@@ -229,12 +230,32 @@ export const makeIoniaProvider: FiatProviderFactory<IoniaMethods> = {
     //
 
     let hiddenCardIds: number[] = asStoreHiddenCards(await store.getItem(STORE_HIDDEN_CARDS_KEY).catch(_ => undefined))
+    let purchaseCardPromise: Promise<IoniaPurchaseCard> // There can only be one!
     const ratesCache: { [currencyCode: string]: { expiry: number; rateQueryPromise: Promise<number> } } = {}
 
     //
     // Private methods:
     //
 
+    async function getPurchaseCard(currencyCode: string, cardAmount: number): Promise<IoniaPurchaseCard> {
+      const newPromise: Promise<IoniaPurchaseCard> = new Promise<IoniaPurchaseCard>(resolve => {
+        // Debounce the request
+        setTimeout(
```

Comment:
This isn't a good debounce implementation as you will still hit the Ionia API on every getQuote. A better implementation is

1. Upon getQuote, save the most recent request amount in local state and set 1s timer
2. If another getQuote comes in, replace the previous request and restart 1s timer
3. Once timer expires, fire off API request
4. If more getQuotes come in, wait until previous API call finishes then go to step 1